### PR TITLE
feat(pii): Allow pii scrubbing expressions for safe fields

### PR DESCRIPTION
**Pull request by jjbayer**
This doesn't simply make matching on arrays work, it enables full expression support for safe fields in the datascrubbing config.

### CODE REVIEW

File: relay-pii/src/convert.rs [hunk 1]
```diff
@@ -1504,4 +1516,36 @@ THd+9FBxiHLGXNKhG/FRSyREXEt+NyYIf/0cyByc9tNksat794ddUqnLOg0vwSkv
...
+                "extra.'do_not_scrub.dot'.**".to_owned(),
+                "$user.id".to_owned(),
+            ],
+            ..simple_enabled_config()
+        })
+        .unwrap();
+
+        let mut pii_processor = PiiProcessor::new(pii_config.compiled());
+        process_value(&mut data, &mut pii_processor, ProcessingState::root()).unwrap();
+        assert_annotated_snapshot!(data);
```
Comment (jjbayer) [inline relay-pii/src/convert.rs]: nit: IMO this snapshot is small enough to inline: 
```suggestion
        assert_annotated_snapshot!(data, @"");
```

File: relay-pii/src/convert.rs [hunk 2]
```diff
@@ -143,9 +143,21 @@ pub fn to_pii_config(
             continue;
         }
 
-        conjunctions.push(SelectorSpec::Not(Box::new(SelectorSpec::Path(vec![
-            SelectorPathItem::Key(field.to_owned()),
-        ]))));
+        let spec = match field.parse() {
+            Ok(spec) => spec,
+            Err(error) => {
+                relay_log::error!(
```
Comment (jjbayer) [inline relay-pii/src/convert.rs:151]:
```suggestion
                // Invalid safe fields should be caught by sentry-side validation. Log an error if they are not.
                relay_log::error!(
```

**Review by jjbayer — COMMENTED** [review]
Looks good! As discussed, let's discuss briefly with the team before merging.

**Review by iker-barriocanal — APPROVED** [review]

**Comment by Dav1dde** [comment]
Re-opened in #2670

**Pull request closed without merging**